### ISSUE TITLE
Slime Spawner bugged in v3.5.5

### ISSUE THREAD

**Issue by BobbyMere** [kangarko/Boss#488]
In version 3.5.5, Slime spawners will not spawn if you set the limit to 1 and there's any other type of mob around it. It works perfectly in 3.4.10 and with vanilla slime spawners, but using boss aslime spawner, nothing will ever spawn. 

I've tested this on two different servers and the problem started with 3.5.5. I would go to an earlier version but 3.4.10 breaks with the latest discordSRV so I'm stuck. 

Edit: It turns out it's not just effecting Slime spawners. If there are two bosses in an area, regardless of mob type, and one has a spawn limit of 1 and the other has a higher limit than 1, the boss with a limit of 1 will never spawn. 

Basically all the bosses in an area are effected by by a spawn limit on a spawner regardless if the mobs fit the type of not. 

This wasn't an issue in v3.4.10

**Comment by kangarko** [comment]
I am sorry, how can I reproduce this problem? I tried setting the spawning limit to 1 and 6 respectively for spawners, all worked fine: https://i.imgur.com/DHmW3kt.png

Please provide me exact instructions to reproduce this issues.

**Comment by kangarko** [comment]
Also please wait till the next version, there were some further changes so please test there.

**Comment by BobbyMere** [comment]
The issue is still persistent in v3.5.6 If one Boss spawner has a limit of 5 and the other spawner has a limit of 1, the spawner with the limit of 5 will override the other one as the boss limit is now effected by all bosses in the area. I can show you directly what I mean if you join 107.175.178.34:53898 or I can give you the world, boss config and coords.

**Comment by BobbyMere** [comment]
world -->[events.zip](https://github.com/kangarko/Boss/files/3871173/events.zip)
Configs --
<img width="626" alt="z" src="https://user-images.githubusercontent.com/25316276/69274018-1a780400-0b9f-11ea-91b7-bffcfa009de3.png">
> [Boss.zip](https://github.com/kangarko/Boss/files/3871176/Boss.zip)
Location -->

**Comment by kangarko** [comment]
From our testing we could not get slimes to spawn at all (tested on a flat map). This is due to the rules of Minecraft. Spawners do not inherit Timed Spawning rules, they behave according to the natural MC rules, so that for. ex. skeleton will not appear in the day.

If you have two spawners, one with 6 limit, the other with 1 limit, but both have the same entity type, the limit will add up, so the other spawner will not spawn. Are your two spawners having different entity types (not Bosses)? https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/entity/EntityType.html

> The Vakiroth boss might spawn when you first get there. But if you kill it, it will never spawn again because there's a mob spawn limit of 1 and the vex's override it

If there is any entity of the same entity type

**Comment by BobbyMere** [comment]
It all worked perfectly in v 3.4.10. I'm fully aware of vanilla limits of slime spawners as it requires a slime chunk for them to spawn (the spawner is in a slime chunk) and skeletons can't spawn during the day(it's underground).

"If you have two spawners, one with 6 limit, the other with 1 limit, but both have the same entity type, the limit will add up, so the other spawner will not spawn. " - Kongarko

Am I wasting my time? I explained this above pretty clearly. They are **NOT** the same entity type. If you have two bosses, REGARDLESS OF ENTITY TYPE, one has a spawn limit of 1 and the other has a spawn limit of 2, the boss with a spawn limit 2 will over ride the spawner with a limit of 1. 

My Vakiroth boss (skeleton type, spawn limit of 1) will NEVER spawn in the area of my vexes (vex type, spawn limit of 12) because the vex's over ride the spawn limiter of the Vakiroth. 

Nothing has changed since I updated from 3.4.10 and it worked perfectly fine in that version. In fact, If I revert back to that version, it works. again. 

Is there a more direct way to communicate? This long wait between responses is eating up time. Should I just look for an alternative plugin and write this one off as a lost?

**Comment by kangarko** [comment]
I apology for the cumbersome communication.

I just tested with a single Vex spawner with your settings of XYZ, and the second Skeleton spawner with your settings for 123. The Skeleton did appear with that cap, the problem is however that we rely on the natural mob spawner rules for spawning. There's no Boss-plugin interference for mob spawners. The only syntetic spawning we do is via the Timed_Spawning option.

I don't understand how the older version of Boss could not have the same problem. Have you changed your server version or anything else when you did test that?

You can still use the older version if you downgrade DiscordSRV to 1.17.x but due to what I wrote above there should be no difference.

I am going to open an issue on Spigot/Paper just to get this confirmed, what's your server version please?

**Comment by BobbyMere** [comment]
My spigot version does not change when I switch out he v3.5.6 for v3.4.10. Did you kill the skeleton one first time it spawned? I can't download discordSRV as other plugins will break 

What about a custom jar of v3..4.10 without the discord dependency, can you give me one?

I'm using spigot 1.13.2 paper

**Comment by BobbyMere** [comment]
I tested version 3.5.1, 3.5.2 and 3.5.3 and the issue starts on v3.5.3

**Comment by BobbyMere** [comment]
[events.zip](https://github.com/kangarko/Boss/files/3883176/events.zip)
[Boss.zip](https://github.com/kangarko/Boss/files/3883177/Boss.zip)
coords: 400 20 -924
with v3.5.3 Vakiroth will not spawn after you kill him once
with v.3.5.2 Vakiroth spawns every 10 seconds

**Comment by kangarko** [comment]
Thanks so much for a detailed feedback and providing me the specific version under which the problem started to occur!

I believe to have found and resolved it, mind giving the next version a try? Thanks!

**Comment by BobbyMere** [comment]
Thank you. It seems to be fixed
